pkg/kubecost/allocationprops.go

```diff
@@ -288,10 +288,16 @@ func (p *AllocationProperties) GenerateKey(aggregateBy []string, labelConfig *La
 				// Indicate that allocation has no services
 				names = append(names, UnallocatedSuffix)
 			} else {
-				// This just uses the first service
-				for _, service := range services {
-					names = append(names, service)
-					break
+				// sometimes load balancers will retain their services when unmounted
+				// so if we have an unmounted pod, make sure the key shows unmounted
+				if p.Pod == UnmountedSuffix {
```

Comment:
Should we be checking more than just the pod? Maybe namespace and container as well?